expression/expression.go

```diff
@@ -208,6 +209,125 @@ func EvalBool(ctx sessionctx.Context, exprList CNFExprs, row chunk.Row) (bool, b
 	return true, false, nil
 }
 
+var (
+	selPool = sync.Pool{
+		New: func() interface{} {
+			return make([]int, 1024)
+		},
+	}
+)
+
+// VecEvalBool does the same thing as EvalBool but it works in a vectorized manner.
+func VecEvalBool(ctx sessionctx.Context, exprList CNFExprs, input *chunk.Chunk, selected, nulls []bool) ([]bool, []bool, error) {
```

Comment:
Will it be promised that there is no gap in input.sel before this line?

Reply:
It is premised now, but we may break this promise later for further optimization.